storage/mysql/tree_storage.go

```diff
@@ -82,6 +87,21 @@ func OpenDB(dbURL string) (*sql.DB, error) {
 		return nil, err
 	}
 
+	// Limit the number of open connections to the database, and periodically
+	// report the currently-open connections as a Prometheus gauge.
```

Comment:
We need to use the `monitoring.MetricsFactory` abstraction to allow for metrics systems other than Prometheus, which (I think) means that `OpenDB` is going to have to change its signature to allow `mf` to be passed in from `newMySQLStorageProvider`.